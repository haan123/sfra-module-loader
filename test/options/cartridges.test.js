/* eslint-disable
  prefer-destructuring,
*/
import webpack from '../helpers/compiler';

describe('Options', () => {
  describe('cartridges', () => {
    beforeEach(() => {
      jest.resetModules()
    });

    test('{Array}', async () => {
      const config = {
        loader: {
          test: /(js)/,
          options: {
            alias: {
              pluggin: 'pluggin',
              core: 'core'
            },
            cartridges: [
              'storefront',
              'pluggin',
              'core'
            ]
          }
        }
      };

      const stats = await webpack('fixture.js', config);
      const [module] = stats.toJson().modules;
      const { source } = module;

      expect(source).toMatchSnapshot();
    });

    test('{String}', async () => {
      const config = {
        loader: {
          test: /(js)/,
          options: {
            alias: {
              pluggin: 'pluggin',
              core: 'core'
            },
            cartridges: 'storefront:pluggin:core'
          }
        }
      };

      const stats = await webpack('fixture.js', config);
      const [module] = stats.toJson().modules;
      const { source } = module;

      expect(source).toMatchSnapshot();
    });

    test('cartridges does not exist', async () => {
      const config = {
        loader: {
          test: /(js)/,
          options: {
            alias: {
              pluggin: 'pluggin',
              core: 'core'
            },
            cartridges: 'abc'
          }
        }
      };

      const stats = await webpack('fixture.js', config);
      const [module] = stats.toJson().modules;
      const { source } = module;

      expect(source).toMatchSnapshot();
    });

  });
});
