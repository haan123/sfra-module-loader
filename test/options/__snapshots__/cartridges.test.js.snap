// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options cartridges {Array} 1`] = `
"module.exports = function(originalModule) {
	if (!originalModule.webpackPolyfill) {
		var module = Object.create(originalModule);
		// module.parent = undefined by default
		if (!module.children) module.children = [];
		Object.defineProperty(module, \\"loaded\\", {
			enumerable: true,
			get: function() {
				return module.l;
			}
		});
		Object.defineProperty(module, \\"id\\", {
			enumerable: true,
			get: function() {
				return module.i;
			}
		});
		Object.defineProperty(module, \\"exports\\", {
			enumerable: true
		});
		module.webpackPolyfill = 1;
	}
	return module;
};
"
`;

exports[`Options cartridges {String} 1`] = `
"module.exports = function(originalModule) {
	if (!originalModule.webpackPolyfill) {
		var module = Object.create(originalModule);
		// module.parent = undefined by default
		if (!module.children) module.children = [];
		Object.defineProperty(module, \\"loaded\\", {
			enumerable: true,
			get: function() {
				return module.l;
			}
		});
		Object.defineProperty(module, \\"id\\", {
			enumerable: true,
			get: function() {
				return module.i;
			}
		});
		Object.defineProperty(module, \\"exports\\", {
			enumerable: true
		});
		module.webpackPolyfill = 1;
	}
	return module;
};
"
`;

exports[`Options cartridges cartridges does not exist 1`] = `
"import core from '*/core';
const pluggin = require('pluggin:pluggin'); // direct request
const dump = require('dump');
const dump2 = require('./dump2');

export default () => {
  core();
  pluggin();
};
"
`;
